feature-libs/product-configurator/rulebased/cpq/rest/converters/cpq-configurator-value-serializer.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Converter } from '@spartacus/core';
+import { Configurator } from '@spartacus/product-configurator/rulebased';
+import { Cpq } from '../cpq.models';
+
+@Injectable()
+export class CpqConfiguratorValueSerializer
+  implements Converter<Configurator.Configuration, Cpq.UpdateValue> {
+  convert(source: Configurator.Configuration): Cpq.UpdateValue {
+    const attribute: Configurator.Attribute = this.findFirstChangedAttribute(
+      source
+    );
+    const updateValue: Cpq.UpdateValue = this.convertAttribute(
+      attribute,
+      source.configId
+    );
+    return updateValue;
+  }
+
+  findFirstChangedAttribute(
```

Comment:
maybe some of the functions can be `protected`?

Reply:
picked

Reply:
Agree, thanks for that hint